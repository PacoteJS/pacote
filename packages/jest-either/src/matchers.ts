export interface AsymmetricMatcher {
  asymmetricMatch(other: any): boolean
}

export function isAsymmetricMatcher(
  matcher: any
): matcher is AsymmetricMatcher {
  return typeof matcher.asymmetricMatch === 'function'
}
