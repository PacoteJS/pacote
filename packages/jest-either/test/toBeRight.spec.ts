import { left, right } from 'fp-ts/lib/Either'
import matchers from '../src'

expect.extend(matchers)

describe('.toBeRight()', () => {
  test('passes when object is a right', () => {
    expect(right('right')).toBeRight()
  })

  test('fails when object is a left', () => {
    expect(() =>
      expect(left('left')).toBeRight()
    ).toThrowErrorMatchingSnapshot()
  })
})

describe('.not.toBeRight()', () => {
  test('passes when object is a left', () => {
    expect(left('left')).not.toBeRight()
  })

  test('fails when object is a right', () => {
    expect(() =>
      expect(right('right')).not.toBeRight()
    ).toThrowErrorMatchingSnapshot()
  })
})
