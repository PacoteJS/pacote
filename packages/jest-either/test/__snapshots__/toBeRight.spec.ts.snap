// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`.not.toBeRight() fails when object is a right 1`] = `
"[2mexpect([22m[31mreceived[39m[2m).not.toBeRight()[22m

Expected Either not to be right, received right."
`;

exports[`.toBeRight() fails when object is a left 1`] = `
"[2mexpect([22m[31mreceived[39m[2m).toBeRight()[22m

Expected Either to be right, received left."
`;
